core/metrics-core-service/src/main/java/org/hawkular/metrics/core/dropwizard/HawkularObjectNameFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2014-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.core.dropwizard;
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+
+import org.hawkular.metrics.core.service.MetricsServiceImpl;
+import org.hawkular.metrics.core.service.log.CoreLogger;
+import org.hawkular.metrics.core.service.log.CoreLogging;
+
+import com.codahale.metrics.ObjectNameFactory;
+
+/**
+ * @author ruben vargas
+ */
+public class HawkularObjectNameFactory implements ObjectNameFactory {
+    private static final CoreLogger LOGGER = CoreLogging.getCoreLogger(MetricsServiceImpl.class);
+    private Map<String, MetaData> metaDataMap = new HashMap<>();
+
+    public HawkularObjectNameFactory(Map<String, MetaData>  map) {
+        this.metaDataMap = map;
+    }
+
+    @Override
+    public ObjectName createName(String type, String domain, String name) {
+        try {
+            Hashtable<String, String> table = new Hashtable<>();
+            table.putAll(metaDataMap.get(name).getTags());
+            ObjectName objectName = new ObjectName(domain, table);
+            return objectName;
+        } catch (MalformedObjectNameException e) {
+            try {
+                return new ObjectName(domain, "name", ObjectName.quote(name));
```

Comment:
Instead of doing this in the Exception category, could we do it some other way? Like detecting which call we're going to need.